src/browser/preload_scripts.ts

```diff
@@ -113,29 +113,29 @@ export function fetchAndLoadPreloadScripts(
 
 
 // resolves to type `PreloadFetched` on success
-// resolves to the input (type `PreloadInstance`) when asset fails to be cached
+// resolves to `undefined` when fetch fails to download asset to Chromium cache
 function fetch(identity: Identity, preloadScript: PreloadInstance): Promise<FetchResponse> {
     return new Promise((resolve: FetchResolver, reject: Rejector) => {
         cachedFetch(identity.uuid, preloadScript.url, (fetchError: null | Error, scriptPath: string | undefined) => {
             if (!fetchError) {
                 resolve({identity, preloadScript, scriptPath});
             } else {
                 updatePreloadState(identity, preloadScript, 'load-failed');
-                reject();
+                resolve();
             }
         });
     });
 }
 
 // resolves to type `PreloadLoaded` on success
-// resolves to the input (type `PreloadFetched`) when fetched asset fails to load (can't be read from cache)
+// resolves to `undefined` when above fetch failed or when successfully fetched asset fails to load from Chromium cache
 function load(opts: FetchResponse): Promise<undefined> {
     return new Promise((resolve: Resolver, reject: Rejector) => {
-        const {identity, preloadScript, scriptPath} = opts;
-
-        if (!scriptPath) {
-            resolve(); // bad fetch but we only get this far if script was optional so resolve
+        if (!opts || !opts.scriptPath) {
+            resolve(); // got fetchError above OR no error but no scriptPath either; in any case don't attempt to load
```

Comment:
We are caching the scripts as a side-effect, the call to `System.setPreloadScript(preloadScript.url, scriptText);`. (We could make this clearer in a comment at top.) Point is resolving early skips the caching call.